### PR TITLE
Imporved Drone Center

### PR DESCRIPTION
**Pull request by RealSilverMoon**
- Fix https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/16127
- Add a internationalization button. It can turn machine name into localized form.
- Add a sort button:
image
Support sort by name, distance and shutdown status.
_The "sort by shutdown status" feature has encountered some issues, causing GUI misalignment in certain cases. This may be attributed to inherent problems within the ModularUI 1 itself._
- Rebalance operating range (T1-128, T2-512, T3-4096). T2 drone is too expensive for a range of 128 that no one would like to use it, while T3 drone should cover the whole base.

### CODE REVIEW

**Comment by RealSilverMoon** [comment]
Also add a sort button:
image
You can sort all machines by name, distance, or if it's shutdown.

File: src/main/java/gregtech/common/tileentities/machines/multi/drone/DroneConnection.java [hunk 1]
```diff
@@ -79,25 +83,42 @@ public boolean reCheckConnection() {
         return isValid();
     }
 
-    public String getCustomName() {
+    public String getCustomName(boolean localized) {
+        if (localized) return GT_LanguageManager.getTranslation("gt.blockmachines." + unlocalizedName + ".name");
         return customName;
     }
 
+    public float getDistance() {
```
Comment (NotAPenguin0) [inline src/main/java/gregtech/common/tileentities/machines/multi/drone/DroneConnection.java:91]: Minor nitpick but I would rename this to `getDistanceSquared()` to better indicate what it actually does

File: src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_DroneCentre.java [hunk 2]
```diff
@@ -178,7 +183,7 @@ protected GT_Multiblock_Tooltip_Builder createTooltip() {
             .addInfo("Monitors multiblock machines in range.")
             .addInfo("Replace maintenance hatch on other multi with drone downlink module.")
             .addInfo("Provides maintenance, power control, monitoring and etc.")
-            .addInfo("Range is determined by drone tier: T1-32, T2-128, T3-512")
+            .addInfo("Range is determined by drone tier: T1-128, T2-512, T3-4096")
```
Comment (NotAPenguin0) [inline src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_DroneCentre.java:186]: Is this an intentional buff to the drone centre or just clarifying values?

File: src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_DroneCentre.java [hunk 3]
```diff
@@ -390,9 +397,9 @@ public List<DroneConnection> getConnectionList() {
 
     public int getRange() {
         return switch (droneLevel) {
-            case 1 -> 32;
-            case 2 -> 128;
-            case 3 -> 512;
+            case 1 -> 128;
```
Comment (NotAPenguin0) [inline src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_DroneCentre.java:400]: Also see above

File: src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_Hatch_DroneDownLink.java [hunk 4]
```diff
@@ -225,10 +226,11 @@ private GT_MetaTileEntity_MultiBlockBase tryFindCoreGTMultiBlock() {
...
             final boolean isMachineBlock;
             tTileEntity = world.getTileEntity(aCoords.posX, aCoords.posY, aCoords.posZ);
-            isMachineBlock = GregTech_API.isMachineBlock(
-                world.getBlock(aCoords.posX, aCoords.posY, aCoords.posZ),
-                world.getBlockMetadata(aCoords.posX, aCoords.posY, aCoords.posZ));
-
+            Block block = world.getBlock(aCoords.posX, aCoords.posY, aCoords.posZ);
+            isMachineBlock = GregTech_API
+                .isMachineBlock(block, world.getBlockMetadata(aCoords.posX, aCoords.posY, aCoords.posZ))
+                || (block == GregTech_API.sBlockReinforced
```
Comment (NotAPenguin0) [inline src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_Hatch_DroneDownLink.java:232]: Add a comment explaining this check for future reference.

**Review by NotAPenguin0 — CHANGES_REQUESTED** [review]
Overall fine, just a few minor details that should probably be looked at before merging

File: src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_DroneCentre.java (hunk 2 above)
Comment (RealSilverMoon) [reply src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_DroneCentre.java:186]: Yes it's a buff. Have seen too many complains about the expensive drones and their narrow range.

File: src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_Hatch_DroneDownLink.java (hunk 4 above)
Comment (RealSilverMoon) [reply src/main/java/gregtech/common/tileentities/machines/multi/drone/GT_MetaTileEntity_Hatch_DroneDownLink.java:234]: done.

**Review by NotAPenguin0 — APPROVED** [review]

**Pull request merged**